hardware/detect.py

```diff
@@ -480,8 +276,10 @@ def main():
     hrdw.append(megacli.detect())
     hrdw.append(diskinfo.detect())
 
-    if not detect_system(hrdw):
+    if not system.detect():
         sys.exit(1)
+    hrdw.append(system.detect())
```

Comment:
This makes a double detection, couldn't we have a single ?

Reply:
yeah, we could assign the result to a variable and check that